packages/serialization/json/src/jsonParseNode.ts

```diff
@@ -11,15 +11,21 @@ export class JsonParseNode implements ParseNode {
 	constructor(private readonly _jsonNode: unknown) {}
 	public onBeforeAssignFieldValues: ((value: Parsable) => void) | undefined;
 	public onAfterAssignFieldValues: ((value: Parsable) => void) | undefined;
-	public getStringValue = () => (typeof this._jsonNode === "string" ? (this._jsonNode as string) : undefined);
-	public getChildNode = (identifier: string): ParseNode | undefined => (this._jsonNode && typeof this._jsonNode === "object" && (this._jsonNode as { [key: string]: any })[identifier] !== undefined ? new JsonParseNode((this._jsonNode as { [key: string]: any })[identifier]) : undefined);
-	public getBooleanValue = () => (typeof this._jsonNode === "boolean" ? (this._jsonNode as boolean) : undefined);
-	public getNumberValue = () => (typeof this._jsonNode === "number" ? (this._jsonNode as number) : undefined);
-	public getGuidValue = () => parseGuidString(this.getStringValue());
-	public getDateValue = () => (this._jsonNode ? new Date(this._jsonNode as string) : undefined);
-	public getDateOnlyValue = () => DateOnly.parse(this.getStringValue());
-	public getTimeOnlyValue = () => TimeOnly.parse(this.getStringValue());
-	public getDurationValue = () => Duration.parse(this.getStringValue());
+	public getStringValue = () => (typeof this._jsonNode === "string" || this._jsonNode === null ? this._jsonNode : undefined);
```

Comment:
basically I thought that this is the place where we do parsing of our json body so that's why added functionality to allow `null` values.
Is it the wrong assumption?

Reply:
Let's try to break this down. The properties of a model need to have null in the type definition so the user can set a null value. This also means that the argument of serialization writer need to accept null as a value so we can properly write it to the serialized representation.
Then we have mostly two choices:
- If the return type of parse node methods does not include null as a valid type it's going to be ok because it's still going to be a subset of what the property we're trying to assign accepts. During the deserialization we can treat null as functionally equivalent to undefined.
- If the return type of the parsnip methods does include null as a valid type then we need to make a difference between the serialization and return null.

The question then becomes do users care about telling the difference between null and undefined when it is returned by the service they are talking to? I would argue they do not since we've never had this request for any other language at this point. I would like to get you and @andrueastman opinion on the matter before we make a final decision on this.

Reply:
> we've never had this request for any other language at this point.
Yeah. I would look at it through this lens. 
It would not be best placed to conceive of requirements here without a solid backing. The working default state as `undefined` would be okay for now and we could always change this once we get solid evidence of the need for distinguishing null and undefined for service level responses.

Reply:
Thanks for chiming in here Andrew. 
@kindest13 are you clear on the modifications to be done at this point or do you have any additional questions?

Reply:
As I understood there's no need to check on `null` value so just adding the `null` type is sufficient.
Although if `getStringValue` received `null` it returns `undefined` then how do we differenciate between `null` and `undefined`?

Reply:
Let me reiterate so we make sure we're all on the same page. First, the models needs to be generated with the type, undefined, null. This is so people can assign null to properties. Then, the serialization writer method parameters need to be able to accept null as well. This is so we can write the new value to the serialized representation. Lastly, the parse node interface does not need to be updated but the implementation needs to treat no from a serialized value as it treats undefined today.
Does this clarify the changes that need to be made?